bridge-fungible-token/src/utils.sw

```diff
@@ -63,12 +63,42 @@ fn shift_decimals_left(bn: U256, d: u8) -> Result<U256, BridgeFungibleTokenError
     Result::Ok(bn_clone)
 }
 
+fn shift_decimals_right(bn: U256, d: u8) -> Result<(U256, u32), BridgeFungibleTokenError> {
```

Comment:
The remainder shouldn't have to be returned. Similar to `overflow` in `shift_decimals_left` we should simply check for it after every division and throw an error if it's ever anything other than 0.

Reply:
I commented below on why I'm currently returning the remainder.

Reply:
But now I see how I can refactor as per your suggestion and still achieve the same end result.